apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-commons/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/commons/interceptor/AbstractMethodInterceptor.java

```diff
@@ -82,11 +82,9 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
             EnhanceRequireObjectCache pathMappingCache = (EnhanceRequireObjectCache) objInst.getSkyWalkingDynamicField();
             String requestURL = pathMappingCache.findPathMapping(method);
             if (requestURL == null) {
-                requestURL = getRequestURL(method);
-                pathMappingCache.addPathMapping(method, requestURL);
-                requestURL = getAcceptedMethodTypes(method) + pathMappingCache.findPathMapping(method);
+                pathMappingCache.addPathMapping(method, getRequestURL(method));
             }
-            operationName = requestURL;
+            operationName = getAcceptedMethodTypes(method) + pathMappingCache.findPathMapping(method);
```

Comment:
> The requestURL is updated after addPathMapping. This means it will use {POST}/url the first time the endpoint is hit and then fall back to /url every time it's hit afterward.

I think you just need move original L86 after L87. The way you changing right now, will make the cache invalid. RIght?

Reply:
I've taken a look at this and moving L87 to L86 is not ideal because of this line:
https://github.com/apache/skywalking/blob/c5972c2b7da417ca6fe2e0e76d1eb5dac77f669c/apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-commons/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/commons/PathMappingCache.java#L52

This appends whatever URL was cached with a method to `classPath`. Meaning a method annotated with `@RequestMapping("/test", method = RequestMethod.GET)` which falls under a class annotated with `@RequestMapping("/case")` will result in the URL `/case/{GET}/test` being cached to the method.

Reply:
My point is, there is a logic error above. execute `pathMappingCache.findPathMapping(method)` twice if cache found.